homeassistant/components/qnap/sensor.py

```diff
@@ -190,106 +176,90 @@
     ),
 )
 
-SENSOR_KEYS: list[str] = [
-    desc.key
-    for desc in (
-        *_SYSTEM_MON_COND,
-        *_CPU_MON_COND,
-        *_MEMORY_MON_COND,
-        *_NETWORK_MON_COND,
-        *_DRIVE_MON_COND,
-        *_VOLUME_MON_COND,
-    )
-]
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_SSL, default=False): cv.boolean,
-        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_MONITORED_CONDITIONS): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_KEYS)]
-        ),
-        vol.Optional(CONF_NICS): cv.ensure_list,
-        vol.Optional(CONF_DRIVES): cv.ensure_list,
-        vol.Optional(CONF_VOLUMES): cv.ensure_list,
-    }
-)
-
 
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the QNAP NAS sensor."""
-    coordinator = QnapCoordinator(hass, config)
+    """Set up the qnap sensor platform from yaml."""
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2023.8.0",
```

Comment:
Based on latest architecture discussion, it seems YAML deprecation is now expected for 6 months not 2.